src/GitHub.App/ViewModels/GitHubPane/PullRequestCheckViewModel.cs

```diff
@@ -0,0 +1,119 @@
+    {
+        private readonly IUsageTracker usageTracker;
+        const string DefaultAvatar = "pack://application:,,,/GitHub.App;component/Images/default_user_avatar.png";
+
+        private string title;
+        private string description;
+        private PullRequestCheckStatusEnum status;
+        private Uri detailsUrl;
+        private string avatarUrl;
+        private BitmapImage avatar;
+
+        public static IEnumerable<IPullRequestCheckViewModel> Build(IViewViewModelFactory viewViewModelFactory, PullRequestDetailModel pullRequest)
+        {
+            return pullRequest.Statuses?.Select(model =>
+            {
+                var statusStateEnum = model.State;
+
+                PullRequestCheckStatusEnum checkStatus;
+                switch (statusStateEnum)
+                {
+                    case StatusStateEnum.Expected:
+                    case StatusStateEnum.Error:
+                    case StatusStateEnum.Failure:
+                        checkStatus = PullRequestCheckStatusEnum.Failure;
+                        break;
+                    case StatusStateEnum.Pending:
+                        checkStatus = PullRequestCheckStatusEnum.Pending;
+                        break;
+                    case StatusStateEnum.Success:
+                        checkStatus = PullRequestCheckStatusEnum.Success;
+                        break;
+                    default:
+                        throw new InvalidOperationException("Unkown PullRequestCheckStatusEnum");
+                }
+
+                var pullRequestCheckViewModel = viewViewModelFactory.CreateViewModel<IPullRequestCheckViewModel>();
+                pullRequestCheckViewModel.Title = model.Context;
+                pullRequestCheckViewModel.Description = model.Description;
+                pullRequestCheckViewModel.Status = checkStatus;
+                pullRequestCheckViewModel.DetailsUrl = new Uri(model.TargetUrl);
+                pullRequestCheckViewModel.AvatarUrl = model.AvatarUrl ?? DefaultAvatar;
+                pullRequestCheckViewModel.Avatar = model.AvatarUrl != null
+                    ? new BitmapImage(new Uri(model.AvatarUrl))
+                    : AvatarProvider.CreateBitmapImage(DefaultAvatar);
+
+                return pullRequestCheckViewModel;
+
+            }) ?? new PullRequestCheckViewModel[0];
+        }
+
+        [ImportingConstructor]
+        public PullRequestCheckViewModel(IUsageTracker usageTracker)
+        {
+            this.usageTracker = usageTracker;
+            OpenDetailsUrl = ReactiveCommand.Create().OnExecuteCompleted(DoOpenDetailsUrl);
+        }
+
+        private void DoOpenDetailsUrl(object obj)
+        {
+            usageTracker.IncrementCounter(x => x.NumberOfPRCheckStatusesOpenInGitHub).Forget();
+        }
+
+        public string Title
+        {
+            get { return title; }
+            set { this.RaiseAndSetIfChanged(ref title, value); }
```

Comment:
Can these properties change? If not then they shouldn't be settable and don't need to raise property changed events.

Reply:
Oh, but I see the point..